src/formats/png/reader.zig

```diff
@@ -0,0 +1,1041 @@
+const std = @import("std");
+const utils = @import("../../utils.zig");
+const png = @import("types.zig");
+const color = @import("../../color.zig");
+const PixelStorage = color.PixelStorage;
+const PixelFormat = @import("../../pixel_format.zig").PixelFormat;
+const Image = @import("../../Image.zig");
+const mem = std.mem;
+const File = std.fs.File;
+const Crc32 = std.hash.Crc32;
+const Allocator = std.mem.Allocator;
+
+// Png specification: http://www.libpng.org/pub/png/spec/iso/index-object.html
+
+pub const any_chunk_type = "_ANY";
+pub const any_chunk_type_id = asU32(any_chunk_type);
+
+pub fn isChunkCritical(id: u32) bool {
+    return (id & 0x20000000) == 0;
+}
+
+inline fn asU32(str: *const [4:0]u8) u32 {
```

Comment:
this function is duplicated into two files, it should be put into a png/utis.zig or something similar.

Or you could put all the chunks in a struct and do something like:

```zig
pub const PngChunks = struct {
   pub const IDAT = chunkIdentifier("IDAT");
   pub const gAMA = chunkIdentifier("gAMA");
};
```